src/lib.rs

```diff
@@ -124,6 +246,317 @@ pub enum DumpError<T> {
 //
 pub type DumpLzss = lzss::Lzss<6, 4, 0x20, { 1 << 6 }, { 2 << 6 }>;
 
+pub fn from_mem(addr: u32, buf: &mut [u8], _meta: bool) -> Result<(), ()> {
```

Comment:
The structures themselves definitely assume 32-bit pointers (an assumption that's pretty baked into both Hubris and Humility). It could be made neutral with respect to address bits, but that's quite a bit more involved.  In terms of testing, though, I want to add a bunch of testing to humpty, which will address the larger point (and can be done without requiring 64-bit addressing by implementing read and write functions that take a 32-bit offset).